app/server/update-controller/src/main/java/io/syndesis/server/update/controller/bulletin/IntegrationUpdateHandler.java

```diff
@@ -142,7 +142,27 @@ private Connection includeConnector(Connection connection) {
     protected List<IntegrationBulletinBoard> compute(ChangeEvent event) {
         final List<IntegrationBulletinBoard> boards = new ArrayList<>();
         final DataManager dataManager = getDataManager();
+
+        /*
+         * Assemble all the artifact collections that are going to be checked in the data manager.
+         * This does it once rather than repeatedly in the various loops since getting stuff from the
+         * DataManager is quite expensive and slow.
+         *
+         * see #3717
+         */
         final List<Integration> integrations = dataManager.fetchAll(Integration.class).getItems();
+        final List<IntegrationDeployment> deployments =
+                dataManager.fetchAll(IntegrationDeployment.class).getItems().stream()
+                .filter(d -> !d.getSpec().isDeleted())
+                .collect(Collectors.toList());
+        final Set<Integration> deployedIintegrations = deployments.stream()
+                .map(d -> d.getSpec())
+                .collect(Collectors.toSet());
+        final List<Connection> connections = dataManager.fetchAll(Connection.class).getItems();
+        final Collection<Integration> allIntegrations = new ArrayList<Integration>();
```

Comment:
the size of the list is known so we can construct it with a proper initial capacity